tf2_py/src/tf2_py.cpp

```diff
@@ -377,13 +377,28 @@ static PyObject *lookupTwistFullCore(PyObject *self, PyObject *args)
 static inline int checkTranslationType(PyObject* o)
 {
   PyTypeObject *translation_type = (PyTypeObject*) PyObject_GetAttrString(pModulegeometrymsgs, "Vector3");
-  return PyObject_TypeCheck(o, translation_type);
+  int type_check = PyObject_TypeCheck(o, translation_type);
+  int attr_check = PyObject_HasAttrString(o, "x") &&
+                   PyObject_HasAttrString(o, "y") &&
+                   PyObject_HasAttrString(o, "z");
+  if (!type_check) {
+    PyErr_WarnEx(PyExc_UserWarning, "translation should be of type Vector3", 1);
```

Comment:
Duck typing is a little bit more pythonic. It seems that this warning would be thrown in known valid cases, and there wouldn't be a way to suppress it thus maybe it shouldn't be  a warning if the duck typing passes?

Reply:
This strict type checking didn't get removed in the end. Don't see a reason for this so would you be open to a PR that gets rid of this type check?

Reply:
If you'd like to improve it that would be appreciated. I'd suggest doing the type check first whickly. And if that doesn't pass do the duck typing adding checks for the type as well as the attribute name.

Reply:
Hmm. Perhaps but I think if passing a Quaternion is going to be much quicker than passing something that quacks like a Quaternion then the user probably _should_ be made aware of that. 

Good point about checking the type of the attributes. Looks like at the moment if you pass something with an 'x' attribute that happens not to be convertible to a double, it gets the value -1.0 thanks to PyFloat_AsDouble